aiidalab/app.py

```diff
@@ -228,12 +240,34 @@ def find_matching_releases(self, specifier, prereleases=None):
         matching_releases.sort(key=parse, reverse=True)
         return matching_releases
 
+    @staticmethod
+    def _strict_dependencies_met(requirements: list[str], python_bin):
+        """Check whether the given requirements are compatible with the core dependencies of a package."""
+        from aiidalab.utils import find_installed_packages
```

Comment:
Yes, because of test, I don't know how to make test work without doing this. Since monkeypatch can only patch object, the `find_installed_packages` import in the file head cannot be overridden. If you know how to make this work it would be great. Here is some information, but I find none of the solution work. 

[1] https://stackoverflow.com/questions/31306080/pytest-monkeypatch-isnt-working-on-imported-function
[2] https://stackoverflow.com/questions/70083364/pytest-monkeypatch-doesnt-apply-to-imported-function

Reply:
I see. Looking at the `find_installed_packages`, what we really want to monkeypatch is the output from `pip list` command. Therefore, I would propose to make a smaller function called `run_pip_list` or something like that, which you then call inside the `find_installed_packages`, and can monkeypatch in tests (maybe?).